tcf_website/views/feedback.py

```diff
@@ -0,0 +1,105 @@
+"""Posting to Feedback"""
+import os
+import json
+import textwrap
+import requests
+from django.http import JsonResponse
+from django.views.decorators.http import require_http_methods
+from django.core.mail import BadHeaderError, send_mail
+
+
+def send_discord(url, message):
+    """Post message to discord server"""
+    data = {"content": message}
+    result = requests.post(url, json=data)
+
+    try:
+        result.raise_for_status()
+    except requests.exceptions.HTTPError:
+        return JsonResponse({'ok': False, 'content': message})
+    else:
+        return JsonResponse({'ok': False, 'content': message})
+
+
+def send_email(subject, message, from_email, recipient_list):
+    """Send email"""
+    try:
+        send_mail(subject, message, from_email, recipient_list)
+    except BadHeaderError:
+        return JsonResponse({'ok': False, 'content': message})
+    else:
+        return JsonResponse({'ok': True, 'content': message})
+
+
+@require_http_methods('POST')
+def discord_feedback(query):
+    """Prepare feedback for discord message"""
+    message_type = query.POST.get("type", "")
+    content = json.loads(query.POST.get("content", ""))
+
+    if message_type == "bug":
+        url = os.environ['DISCORD_URL_BUG']
+        message = textwrap.dedent(f"""\
+            Bug Found!
+            **URL:** {content["url"]}
+            **Categories:** {content["categories"]}
+            **Email:** {content["email"]}
+            **Description:** {content["description"]}""")
+        return send_discord(url, message)
+
+    if message_type == "feedback":
+        url = os.environ['DISCORD_URL_FEEDBACK']
+        message = textwrap.dedent(f"""\
+            Feedback submitted
+            **Name:** {content["fname"]} {content["lname"]}
+            **Email:** {content["email"]}
+            **Title:** {content["title"]}
+            **Message:** {content["message"]}""")
+        return send_discord(url, message)
+    return JsonResponse({'ok': False, 'content': query})
+
+
+@require_http_methods('POST')
+def email_feedback(query):
+    """Prepare feedback for email"""
+    message_type = query.POST.get("type", "")
+    content = json.loads(query.POST.get("content", ""))
+
+    from_email = os.environ['EMAIL_HOST_USER']
+    to_email = content["email"]
+    recipient_list = [
+        to_email,
+        from_email
+    ]
+    subject = "[theCourseForum] Thank you for your feedback!"
+
+    if message_type == "bug":
```

Comment:
It's probably fine for now, but seems a little clunky to have the email template hard coded if we need to make edits in the future.